.github/workflows/preview-env-deploy.yml

```diff
@@ -28,6 +32,18 @@ jobs:
             secret/data/products/camunda-docs/ci/preview-environment PREVIEW_ENV_BUCKET_NAME;
             secret/data/products/camunda-docs/ci/preview-environment PREVIEW_ENV_GCLOUD_SA_KEY;
 
+      - name: Restore docusaurus cache
+        if: github.ref != 'refs/heads/main'
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            ${{ github.workspace }}/.docusaurus
+            ${{ github.workspace }}/**/.cache
+          key: |
+            ${{ runner.os }}-docusaurus-preview-env-${{ hashFiles('**/package-lock.json', '**/npm-shrinkwrap.json', '**/yarn.lock', '**/pnpm-lock.yaml') }}
```

Comment:
I'm not super familiar with this docusaurus project, how did you come up with the part `${{ hashFiles('**/package-lock.json', '**/npm-shrinkwrap.json', '**/yarn.lock', '**/pnpm-lock.yaml') }}` and are all of those needed?

Reply:
The inspiration this mechanism came from the already existent action which [caches docusaurus builds](https://github.com/docuactions/cache/blob/main/action.yml). I am under the impression that  these files dont get cached but are used in order to generate a unique key for the hash